pkg/ingester/instance.go

```diff
@@ -734,7 +736,9 @@ func sendBatches(ctx context.Context, i iter.EntryIterator, queryServer QuerierQ
 
 		stats.AddIngesterBatch(int64(batchSize))
 		batch.Stats = stats.Ingester()
-
+		if isDone(ctx) {
+			break
+		}
 		if err := queryServer.Send(batch); err != nil {
 			return err
```

Comment:
If the intention is that context cancellation should never result in an error return from this function then I would think you'd want to swallow that here too. (and in similar places)